docs/developer-guide/creating-agents.md

```diff
@@ -0,0 +1,763 @@
+            content=combined_analysis,
+            metadata={
+                "modalities": list(results.keys()),
+                "image_processed": bool(record.image_url),
+                "text_processed": bool(record.content)
+            }
+        )
+```
+
+### 4. Tool-Enabled Agent
+
+For agents that can use external tools:
+
+```python
+from buttermilk._core.mcp_decorators import tool
+from buttermilk._core.tool_definition import AgentToolDefinition
+
+class ToolEnabledAgent(Agent):
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.calculator = Calculator()
+        self.web_search = WebSearch()
+    
+    @tool(
+        name="calculate",
+        description="Perform mathematical calculations",
+        parameters={
+            "expression": {"type": "string", "description": "Mathematical expression"}
+        }
+    )
+    async def calculate(self, expression: str) -> dict:
+        """Calculate a mathematical expression."""
+        try:
+            result = self.calculator.evaluate(expression)
+            return {"result": result, "expression": expression}
+        except Exception as e:
+            return {"error": str(e), "expression": expression}
+    
+    @tool(
+        name="search_web",
+        description="Search the web for information",
+        parameters={
+            "query": {"type": "string", "description": "Search query"}
+        }
+    )
+    async def search_web(self, query: str) -> dict:
+        """Search the web for information."""
+        results = await self.web_search.search(query)
+        return {
+            "query": query,
+            "results": results[:5],  # Top 5 results
+            "total_found": len(results)
+        }
+    
+    async def _process(self, *, record: Record, **kwargs) -> AgentOutput:
+        # This agent can use tools during processing
+        content = record.content
+        
+        # Example: Use tools if needed
+        if "calculate" in content.lower():
+            # Extract expression and calculate
+            expr = self._extract_expression(content)
+            calc_result = await self.calculate(expr)
+            response = f"Calculation result: {calc_result}"
+        elif "search" in content.lower():
+            # Extract query and search
+            query = self._extract_query(content)
+            search_result = await self.search_web(query)
+            response = f"Search results: {search_result}"
+        else:
+            response = f"Processed: {content}"
+        
+        return AgentOutput(
+            agent_name=self.name,
+            content=response,
+            metadata={"tools_used": ["calculate", "search_web"]}
+        )
+```
+
+## Configuration
+
+### Agent Configuration Files
+
+```yaml
+# conf/agents/my_agent.yaml
+name: "content_analyzer"
+type: "ContentAnalyzer"
+role: "ANALYZER"
+description: "Analyzes content for sentiment and bias"
+
+parameters:
+  model: "gemini-pro"
```

Comment:
this is incorrect, we configure models at the flow level usually. All these examples should instead reference the example flow configuration and example agents -- trans, tox, osb, zot for flows, and judge, synth, diff, spy, structuredllmhost, research assistant for agents. Cut out the fake configs.